pkg/ring/lifecycler.go

```diff
@@ -601,6 +624,9 @@ func (i *Lifecycler) heartbeat(ctx context.Context) {
 	i.lifecyclerMetrics.consulHeartbeats.Inc()
 	ctx, cancel := context.WithTimeout(ctx, i.cfg.HeartbeatPeriod)
 	defer cancel()
+	if i.delegate != nil {
```

Comment:
Instead of explicitly checking `i.deletegate != nil`, can we define a noopDelegate as the default which does nothing?